comp/core/workloadmeta/collectors/internal/containerd/containerd.go

```diff
@@ -279,13 +279,28 @@ func (c *collector) notifyInitialImageEvents(ctx context.Context, namespace stri
 		return err
 	}
 
+	filteredImages := make(map[workloadmeta.EntityID]*workloadmeta.ContainerImageMetadata)
```

Comment:
IMO these images are not filtered but `merged`, maybe we should rename the variable to reflect that.

Note: IMO this is a common problem in workloadmeta collectors where we need to merge entities coming from the same source, we might benefit from a generic solution but it's out of scope for this change